src/expr.cpp

```diff
@@ -1121,16 +1124,17 @@ llvm::Value *UnaryExpr::GetValue(FunctionEmitContext *ctx) const {
     if (expr == nullptr)
         return nullptr;
 
+    const bool isSigned = !expr->GetType()->IsUnsignedType();
```

Comment:
The negation of `IsUnsignedType()` should not be used. It will give incorrect result for floating point types, as the are not unsigned.

Instead, `IsSignedType()` needs to be introduced, which specifically checks that it's signed integer type.